students/MicahBraun/Lesson 04/Markov_Analysis.py

```diff
@@ -0,0 +1,100 @@
+
+#   --------------------------------------------     FUNCTIONS     -----------------------------------------------------
+
+
+def process_f(filename, order=3):
+    """Reads-in file contents (from filename) with default value of 3 (trigram)"""
+
+    fp = open(filename)                             # open file
+
+    for line in fp:                                 # for each line of text in file: do something
+        for word in line.rstrip().split():          # for each word in each line: strip \r and split into tuple
+            process_word(word, order)               # for tuple, call process word()
+    fp.close()                                      # close file
+
+
+def process_word(word, order=3):
+    """Processes each tuple/word (str, int)
+    "During the first few iterations, all we do is store up the words;   <--- This quote is from Think Python's Example.
+    after that we start adding entries to the dictionary."               <--- I liked their description better than
+                                                                         <--- anything I could come up with.
+    """
+    global root_word                                       # access global tuple word collection
+    if len(root_word) < order:                             # if the length of the tuple is less than order(3), add-
+        root_word += (word,)                               # -word to root_word tuple. it needs to be at least 3 to work
+        return
+
+    try:
+        potential_words_map[root_word].append(word)        # add value of root_word to dict potential_words_map{}
+    except KeyError:
+        potential_words_map[root_word] = [word]            # if key of root_word does not already exist in dict, create
+
+    root_word = arrange_trigram(root_word, word)          # call arrange_trigram for root_word, word
+
+
+def randomize(n=300):
+    """Generates random words from the analyzed text -- n: number of words to generate"""
+    global big_list                                                 # accesses global var big_list
+    start = random.choice(list(potential_words_map.keys()))         # a random value (word) from dict
+
+    for i in range(n):                                              # for items in range 0-300
+        potential_words = potential_words_map.get(start, None)      # potential_words = random
+        if potential_words == None:                                 # if == None, randomize(300 - 300)
+            randomize(n-i)                                          # randomize all
+            return                                              # Not None
+        word = random.choice(potential_words)                       # word = random selection from potential_words
+        big_list.append(word + " ")                                 # append word to big_list along with whitespace
+        start = arrange_trigram(start, word)                       # start = call arrange_trigram() on start, word
+
```

Comment:
I don't think this is accomplishing quite what the assingment specifies. The algorithm should begin with a random word pair, append that pair to the new text (it's good to use a list at this point as you're doing), then choose a random following word from the dict to append to the new text. The next iteration should take the last two word pair from the new text and use them to look up another random following word to append and so on